File: lesson 2023-09-24/array.pas
program hello;
var
a:array[1..600] of integer;
i,N,max1,max2:integer;

begin
  read(N);
  Randomize;
  for i := 1 to N do
    a[i] := Random(601) - 300;

  max1 := -999;
  max2 := -999;

  for i := 1 to N do
  begin
    if (a[i] mod 3 = 0) then
    begin
      if (a[i] > max1) then
      begin
        max2 := max1;
        max1 := a[i];
      end
      else if (a[i] > max2) then
        max2 := a[i];
    end;
  end;

  writeln('Массив:');
  for i := 1 to N do
    write(a[i], ' ');

  writeln;
  writeln('max1: ', max1);
  writeln('max2: ', max2);
end.

File: lesson 2023-09-24/array.py
import random


def find_max_multiples_of_three(arr):
    max1 = -300
    max2 = -300

    for num in arr:
        if num % 3 == 0:
            if num > max1:
                max2 = max1
                max1 = num
            elif num > max2:
                max2 = num

    return max1, max2

N = 10
arr = [random.randint(-300, 300) for _ in range(N)]

max1, max2 = find_max_multiples_of_three(arr)

print("Массив:")
print(arr)
print("Первый максимальный кратный 3:", max1)
print("Второй максимальный кратный 3:", max2)
